src/mmw/js/src/modeling/models.js

```diff
@@ -837,6 +838,13 @@ var ScenarioModel = Backbone.Model.extend({
         return taskModel.start(taskHelper);
     },
 
+    fetchResultsAfterInitial: function() {
+        var self = this;
+        this.fetchResultsIfNeeded().done(function() {
+            self.debouncedFetchResults();
```

Comment:
Is there a reason to prefer this indirection? Since this anonymous function does only one thing: execute `debouncedFetchResults()`, we could give it directly to `.done()`, as shown in the comment above.